.browserslistrc

```diff
@@ -0,0 +1,2 @@
+last 2 version
+not dead
```

Comment:
What made you choose this list instead of `defaults` (`> 0.5%, last 2 versions, Firefox ESR, not dead`)?

Reply:
Ah, didn't know about `defaults`. On the other hand defaults contains for example "safari 5.1". I'm pretty sure klaro does not work on that. Wouldn't it be nice to look at the .browserslistrc and see what browsers are actually supported?

Reply:
It would be nice to see a list of supported browsers by looking at the file, however, aliases like `> 2%` and `last 2 versions` also include non-Western browsers that we probably wouldn't know about. Browserlist mentions high usage of QQ Browser in China and Opera Mini in Africa.

I would want to trust browserlist with one or more of those aliases because the project includes this expert knowledge.

https://github.com/browserslist/browserslist#best-practices

Reply:
I tought about this again. There is a way to add polyfills automatically (`useBuiltIns: 'usage'` in babel.config.js). I didn't do this because i had some bad experiences with this in the past. It will add another 7 KB to the bundle. But it respects the browserlist and may work on more browsers (e.g. Android).

```
Using polyfills with `usage` option:

[klaro/src/klaro.js] Added following core-js polyfills:
  es.array.concat { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.map { "android":"76", "ie":"11" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/components/app.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.object.get-prototype-of { "android":"76", "ie":"11" }
  es.object.set-prototype-of { "android":"76" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/consent-manager.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.filter { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.array.map { "android":"76", "ie":"11" }
  es.function.name { "ie":"11" }
  es.object.keys { "android":"76", "ie":"11" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.regexp.constructor { "android":"76", "edge":"17", "ie":"11" }
  es.regexp.to-string { "android":"76", "edge":"17", "ie":"11" }
  es.set { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  es.string.replace { "android":"76", "edge":"17", "firefox":"68", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }
  web.dom-collections.for-each { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/utils/maps.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.map { "android":"76", "ie":"11" }
  es.object.keys { "android":"76", "ie":"11" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/utils/i18n.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.concat { "android":"76", "ie":"11" }
  es.array.from { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.array.join { "ie":"11" }
  es.array.slice { "android":"76", "ie":"11" }
  es.map { "android":"76", "ie":"11" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.parse-int { "android":"76" }
  es.regexp.constructor { "android":"76", "edge":"17", "ie":"11" }
  es.regexp.to-string { "android":"76", "edge":"17", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  es.string.match { "android":"76", "edge":"17", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/translations/index.js] Based on your code and targets, core-js polyfills were not added.

[klaro/src/components/consent-notice.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.array.join { "ie":"11" }
  es.array.map { "android":"76", "ie":"11" }
  es.object.get-prototype-of { "android":"76", "ie":"11" }
  es.object.set-prototype-of { "android":"76" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/utils/cookies.js] Added following core-js polyfills:
  es.function.name { "ie":"11" }
  es.regexp.constructor { "android":"76", "edge":"17", "ie":"11" }
  es.regexp.to-string { "android":"76", "edge":"17", "ie":"11" }
  es.string.split { "android":"76", "edge":"17", "ie":"11" }

[klaro/src/components/consent-modal.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.object.get-prototype-of { "android":"76", "ie":"11" }
  es.object.set-prototype-of { "android":"76" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/utils/config.js] Added following core-js polyfills:
  es.array.from { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.set { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/components/apps.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.filter { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.array.map { "android":"76", "ie":"11" }
  es.function.name { "ie":"11" }
  es.object.assign { "android":"76", "ie":"11" }
  es.object.get-prototype-of { "android":"76", "ie":"11" }
  es.object.set-prototype-of { "android":"76" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }

[klaro/src/components/icons.js] Based on your code and targets, core-js polyfills were not added.

[klaro/src/components/app-item.js] Added following core-js polyfills:
  es.symbol { "android":"76", "ie":"11" }
  es.symbol.description { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "samsung":"9.2" }
  es.symbol.iterator { "android":"76", "ie":"11" }
  es.array.iterator { "android":"76", "ie":"11", "opera":"46" }
  es.array.join { "ie":"11" }
  es.array.map { "android":"76", "ie":"11" }
  es.function.name { "ie":"11" }
  es.object.assign { "android":"76", "ie":"11" }
  es.object.get-prototype-of { "android":"76", "ie":"11" }
  es.object.set-prototype-of { "android":"76" }
  es.object.to-string { "android":"76", "ie":"11" }
  es.string.iterator { "android":"76", "ie":"11" }
  web.dom-collections.iterator { "android":"76", "edge":"17", "ie":"11", "ios":"12.2", "opera":"46", "safari":"12.1" }
```

Reply:
I prefer the automatic detection, especially because it allows to target other browsers without changing the code.
A gain of 7 kb is quite a lot for a tool which praised itself being less than 20 kb (zipped) at some point.

However, people already asked to exclude translations they don't need... so I'm willing to accept this, with the possibility of a website offering custom building in the future.

Reply:
Changed with the latest commit